diff -up SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h.fix SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h
--- SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h.fix	2016-02-05 12:18:29.514348406 -0500
+++ SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h	2016-02-05 12:18:45.898990489 -0500
@@ -79,6 +79,7 @@ void SDL_WAYLAND_UnloadSymbols(void);
 #define wl_proxy_get_user_data (*WAYLAND_wl_proxy_get_user_data)
 #define wl_proxy_add_listener (*WAYLAND_wl_proxy_add_listener)
 #define wl_proxy_marshal_constructor (*WAYLAND_wl_proxy_marshal_constructor)
+#define wl_proxy_marshal_constructor_versioned (*WAYLAND_wl_proxy_marshal_constructor_versioned)
 
 #define wl_seat_interface (*WAYLAND_wl_seat_interface)
 #define wl_surface_interface (*WAYLAND_wl_surface_interface)
diff -up SDL2-2.0.4/src/video/wayland/SDL_waylandsym.h.fix SDL2-2.0.4/src/video/wayland/SDL_waylandsym.h
--- SDL2-2.0.4/src/video/wayland/SDL_waylandsym.h.fix	2016-07-10 17:55:53.970424955 +0100
+++ SDL2-2.0.4/src/video/wayland/SDL_waylandsym.h	2016-07-10 18:03:01.627109626 +0100
@@ -55,6 +55,9 @@ SDL_WAYLAND_SYM(void, wl_list_insert_lis
 SDL_WAYLAND_MODULE(WAYLAND_CLIENT_1_4)
 SDL_WAYLAND_SYM(struct wl_proxy *, wl_proxy_marshal_constructor, (struct wl_proxy *, uint32_t opcode, const struct wl_interface *interface, ...))
 
+SDL_WAYLAND_MODULE(WAYLAND_CLIENT_1_10)
+SDL_WAYLAND_SYM(struct wl_proxy *, wl_proxy_marshal_constructor_versioned, (struct wl_proxy *proxy, uint32_t opcode, const struct wl_interface *interface, uint32_t version, ...))
+
 SDL_WAYLAND_INTERFACE(wl_seat_interface)
 SDL_WAYLAND_INTERFACE(wl_surface_interface)
 SDL_WAYLAND_INTERFACE(wl_shm_pool_interface)
